src/routing_table/mod.rs

```diff
@@ -753,46 +724,53 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
     ///       this group; otherwise
     ///     - returns `Err(Error::CannotRoute)`
     pub fn targets(&self,
-                   dst: &Destination<T>,
+                   dst: &Authority<T>,
                    exclude: T,
                    route: usize)
                    -> Result<HashSet<T>, Error> {
-        let (closest_group, target_name) = match *dst {
-            Destination::Group(ref target_name) => {
-                let (prefix, group) = self.closest_group(target_name);
-                if *prefix == self.our_group_prefix {
-                    return Ok(group.clone());
+        let target_name = dst.name();
+        let (closest_section, target_name) = if dst.is_section() || dst.is_group() {
```

Comment:
We set `target_name` to `target_name` on both brances now, I think, so it doesn't need to be assigned here.